### PR TITLE
API guidance updates

### PR DESCRIPTION
**Pull request by cwrw**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-3378

updating the guidance for providers for MVP

### Changes proposed in this pull request

- Add new pages to guidance
- Update content
- Update styling for code blocks

### Failed feature specs screenshots

If any of the feature specs would fail, there will be page on the wiki with all
failures:

https://github.com/DFE-Digital/npq-registration/wiki/

### CODE REVIEW

File: app/views/api/guidance/index.html.erb [hunk 1]
```diff
@@ -45,27 +40,46 @@
...
-          right amount by the DfE</p>
-        <p class="govuk-body">
-          <%= govuk_link_to "Browse guidance", api_guidance_path, class: "govuk-!-font-weight-bold" %>
-        </p>
+    <div class="govuk-grid-column-full-from-desktop">
+      <hr class="govuk-section-break govuk-section-break--xl govuk-section-break--invisible">
+      <div class="govuk-grid-column-two-thirds ">
+        <h2 class="govuk-heading-l">How-to guides</h2>
+        <p class="govuk-body">Guidance on how lead providers can integrate with this API so they can view, submit and update data.</p>
+        <ul class="govuk-list">
```
Comment (cwrw) [inline app/views/api/guidance/index.html.erb:48]: do we want bullets here?

File: app/views/api/guidance/index.html.erb [hunk 2]
```diff
@@ -123,16 +138,14 @@
...
-        <h2 class="govuk-heading-l">Support</h2>
-        <p class="govuk-body">
-          If you have any questions or suggestions, email us at
-          continuing-professional-development@digital.education.gov.uk.
-        </p>
+        <h2 class="govuk-heading-l">Community</h2>
         <p class="govuk-body">
-          We'll try to get back to you within 2 working days.
+          We're always working to improve this API so it meets the needs of lead providers. You can help us by sharing
+          your ideas with us on our dedicated <a href="#" class="govuk-link">Slack channel</a>.
```
Comment (cwrw) [inline app/views/api/guidance/index.html.erb:144]: we don't have a slack channel?

File: app/views/api/guidance/get-started.md [hunk 3]
```diff
@@ -2,15 +2,69 @@
 
 ## Connect to the API
 
-Connect to the API by integrating local provider CRM systems with it. A unique authentication token is needed to
-connect to the API. Each token is associated with a single provider and will give providers access to appropriate CPD
-participant data. Authentication tokens do not expire.
+Providers integrate their local CRM systems by connecting to this API's test environments.
 
-## Production environment
+We'll add information about connecting to the production environments when they go live in November 2024.
```
Comment (cwrw) [inline app/views/api/guidance/get-started.md:7]: should we say when they connect?

File: app/views/api/guidance/test-environments.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+## What are the test environments
+
+The test environments are used to test API integrations without affecting real data.
```
Comment (cwrw) [inline app/views/api/guidance/test-environments.md:3]: do we want to add more detail here around bearer tokens or ECF testing envs? might be a repeat of existing content

File: app/views/api/guidance/how-npqs-work.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# How NPQs work  
+
+## The application stage
+
+1. Once someone has registered with us to do an NPQ, providers can view their application data via our API.   
+2. Providers complete their own suitability and application processes.
+3. Providers process applications via our API. This is where providers can confirm whether or not training will be DfE-funded.
+4. Process set successful applicants up on their training portals.
```
Comment (cwrw) [inline app/views/api/guidance/how-npqs-work.md:8]: providers?

File: app/views/api/guidance/view-accept-or-reject-applications.md [hunk 6]
```diff
@@ -0,0 +1,366 @@
+# View, accept or reject applications
+
```
Comment (cwrw) [inline app/views/api/guidance/view-accept-or-reject-applications.md:2]: do we need an extra title here?

File: app/views/api/guidance/view-accept-or-reject-applications.md [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+### Example request body
+
+```
+{
+  "data": {
+    "type": "npq-application-accept",
+    "attributes": {
+      "schedule_identifier": "npq-leadership-spring"
+      "funded_place": null
```
Comment (cwrw) [inline app/views/api/guidance/view-accept-or-reject-applications.md:158]: do we want to say anything about funded place?

File: app/views/api/guidance/data-states.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+      <th scope="row" class="govuk-table__header"><code>application</code></th>
+      <td class="govuk-table__cell">The application a person makes to be trained on an NPQ course. Applications include funding details</td>
+    </tr>
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>participant</code></th>
+      <td class="govuk-table__cell">A person registered for an NPQ course</td>
+    </tr>
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>schedule</code></th>
+      <td class="govuk-table__cell">The expected timeframe in which a participant will complete their NPQ course. Schedules include defined [milestone dates](ADD LINK) against which DfE validates the declarations submitted by providers</td>
```
Comment (cwrw) [inline app/views/api/guidance/data-states.md:23]: add link?

File: app/views/api/guidance/data-states.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+  <tbody class="govuk-table__body">
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>active</code></th>
+      <td class="govuk-table__cell govuk-table__cell">Participants currently in training</td>
+      <td class="govuk-table__cell govuk-table__cell">Update participant data and submit declarations</td>
+    </tr>
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>deferred</code></th>
+      <td class="govuk-table__cell govuk-table__cell">Participants who've deferred training</td>
+      <td class="govuk-table__cell govuk-table__cell">Notify DfE when the participant [resumes training](ADD LINK)</td>
```
Comment (cwrw) [inline app/views/api/guidance/data-states.md:120]: add link?

File: app/views/api/guidance/what-you-can-do-in-the-api.md [hunk 10]
```diff
@@ -1,46 +1,48 @@
...
 
-- View all applications
-- View a specific application
-- Accept an application
-- Reject an application
-- Update an application due to a change in circumstance
+### Applications
 
-## View and update participant data
+1. A person submits an application for a course using the [Register for a national professional qualification service](ADD LINK).
```
Comment (cwrw) [inline app/views/api/guidance/what-you-can-do-in-the-api.md:13]: add link?

File: app/views/api/guidance/view-and-update-participant-data.md [hunk 11]
```diff
@@ -0,0 +1,363 @@
...
+
+<div class="govuk-inset-text">
+Providers can filter results by adding updated_since filters to the parameter. For example: GET /api/v{n}/participants/ecf?filter[updated_since]=2020-11-13T11:21:55Z
+</div>
+
+### Check for participant ID changes
+
+Providers can check if an NPQ participant’s ID has changed using the `participant_id_changes` nested structure in the `NPQEnrolment`, which contains a `from_participant_id` and a `to_participant_id` string fields, as well a `changed_at` date value.
+
+### Duplicate IDs
```
Comment (cwrw) [inline app/views/api/guidance/view-and-update-participant-data.md:25]: this seems like a duplicate to the below?

File: app/views/api/guidance/view-and-update-participant-data.md [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+GET /api/v3/participants/npq/outcomes
+```
+
+Participants can either pass or fail assessment at the end of their NPQ course. These outcomes are submitted by providers within ```completed``` declaration submissions.
+
+<div class="govuk-inset-text">
+Outcomes are sent to the Database of Qualified Teachers (DQT). They issue certificates to participants who've passed.
+</div>
+
+Successful requests will return a response body including an outcome ```state``` value to signify:
```
Comment (cwrw) [inline app/views/api/guidance/view-and-update-participant-data.md:264]: possible duplication in outcomes?

File: app/views/api/guidance/what-you-can-do-in-the-api.md [hunk 13]
```diff
@@ -1,46 +1,48 @@
-[//]: # (Placeholder for What you can do in the API)
```
Comment (cwrw) [inline app/views/api/guidance/what-you-can-do-in-the-api.md:1]: I think we can remove this?

File: app/views/api/guidance/glossary.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
+# Glossary
```
Comment (cwrw) [inline app/views/api/guidance/glossary.md:1]: should this remain, we don't link to this?

File: app/views/api/guidance/error-messages.md [hunk 15]
```diff
@@ -0,0 +1,181 @@
+# Error messages explained
```
Comment (cwrw) [inline app/views/api/guidance/error-messages.md:1]: no link to this?

Comment (cwrw) [reply app/views/api/guidance/error-messages.md:1]: move into v3 branc

File: app/views/api/guidance/glossary.md (hunk 14 above)
Comment (cwrw) [reply app/views/api/guidance/glossary.md:1]: move to v3

File: app/views/api/guidance/api-requests.md [hunk 16]
```diff
@@ -0,0 +1,1037 @@
+# API requests
```
[the remark this answers is not in the document]
Comment (cwrw) [reply app/views/api/guidance/api-requests.md:1]: remove

File: app/views/api/guidance/what-you-can-do-in-the-api.md (hunk 13 above)
Comment (cwrw) [reply app/views/api/guidance/what-you-can-do-in-the-api.md:1]: remove

File: app/views/api/guidance/view-and-update-participant-data.md (hunk 12 above)
Comment (cwrw) [reply app/views/api/guidance/view-and-update-participant-data.md]: intentional

File: app/views/api/guidance/view-and-update-participant-data.md (hunk 11 above)
Comment (cwrw) [reply app/views/api/guidance/view-and-update-participant-data.md:25]: intentional

File: app/views/api/guidance/data-states.md (hunk 9 above)
Comment (cwrw) [reply app/views/api/guidance/data-states.md:120]: add link to resume swagger

File: app/views/api/guidance/data-states.md (hunk 8 above)
Comment (cwrw) [reply app/views/api/guidance/data-states.md:23]: add link to milestone page in guidance

File: app/views/api/guidance/view-accept-or-reject-applications.md (hunk 7 above)
Comment (cwrw) [reply app/views/api/guidance/view-accept-or-reject-applications.md]: agreed to hold off for now and update if required

File: app/views/api/guidance/view-accept-or-reject-applications.md (hunk 6 above)
Comment (cwrw) [reply app/views/api/guidance/view-accept-or-reject-applications.md:2]: intentional

File: app/views/api/guidance/index.html.erb (hunk 1 above)
Comment (cwrw) [reply app/views/api/guidance/index.html.erb:48]: keep as is

File: app/views/api/guidance/index.html.erb (hunk 2 above)
Comment (cwrw) [reply app/views/api/guidance/index.html.erb:144]: on your dedicated Slack channel.

File: app/views/api/guidance/test-environments.md (hunk 4 above)
Comment (cwrw) [reply app/views/api/guidance/test-environments.md:5]: leave as is, might revisit in future

File: app/views/api/guidance/data-states.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+  <tbody class="govuk-table__body">
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>active</code></th>
+      <td class="govuk-table__cell govuk-table__cell">Participants currently in training</td>
+      <td class="govuk-table__cell govuk-table__cell">Update participant data and submit declarations</td>
+    </tr>
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header"><code>deferred</code></th>
+      <td class="govuk-table__cell govuk-table__cell">Participants who've deferred training</td>
+      <td class="govuk-table__cell govuk-table__cell">Notify DfE when the participant [resumes training](https://npq-registration-review-1388-web.test.teacherservices.cloud/api/docs/v3#/NPQ%20Participants/put_api_v3_participants_npq__id__resume)</td>
```
[the remark this answers is not in the document]
Comment (cwrw) [reply app/views/api/guidance/data-states.md:120]: ah my bad will fix this

[the remark this answers is not in the document]
Comment (cwrw) [reply app/views/api/guidance/data-states.md]: addressed in https://github.com/DFE-Digital/npq-registration/pull/1388/commits/867f7f5bb013f24a0a3c6744b50d705b8502a6fc

File: app/views/layouts/api_docs.html.erb [hunk 18]
```diff
@@ -7,7 +7,10 @@
 <html lang="en" class="govuk-template govuk-template--wide">
   <%= render partial: "layouts/shared/head" %>
 
-  <body class="govuk-template__body">
+  <body class="govuk-template__body js-enabled <%= yield :body_class %>">
```
Comment (ethax-ross) [inline app/views/layouts/api_docs.html.erb:10]: Worth bunging this in a partial as we use it in multiple places?

**Review by ethax-ross — APPROVED** [review]
Looks great 👏

**Review by ethax-ross — APPROVED** [review]

File: app/views/api/guidance/index.html.erb [hunk 19]
```diff
@@ -45,27 +40,46 @@
...
+
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full-from-desktop">
+      <hr class="govuk-section-break govuk-section-break--xl govuk-section-break--visible">
+      <div class="govuk-grid-column-two-thirds ">
+        <h2 class="govuk-heading-l">Schedules and milestone dates</h2>
+        <p class="govuk-body">Details of the contractual retention periods during which providers must submit relevant declarations evidencing training delivery and participant retention.</p>
+        <ul class="govuk-list">
+          <li><%= govuk_link_to("View NPQ, early headship coaching offer and addtional support offer schedules", api_guidance_page_path(page: "schedules-and-milestone-dates"), class: "govuk-!-font-weight-bold") %></li>
```
Comment (leandroalemao) [inline app/views/api/guidance/index.html.erb:81]: *additional

File: app/views/api/guidance/schedules-and-milestone-dates.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+      <td class="govuk-table__cell">1 October 2022</td>
+    </tr>
+    <tr class="govuk-table__row">
+      <th scope="row" class="govuk-table__header">From 1 January 2023</th>
+      <td class="govuk-table__cell">1 January 2023</td>
+    </tr>
+  </tbody>
+</table>
+
+## Early headship coaching offer (EHCO) and addtional support offer (ASO) schedules
```
Comment (leandroalemao) [inline app/views/api/guidance/schedules-and-milestone-dates.md:144]: *additional

File: app/views/api/guidance/view-and-update-participant-data.md [hunk 21]
```diff
@@ -0,0 +1,363 @@
...
+            "changed_at": "2023-09-23T02:22:32.000Z",
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+## Defer a particpant
```
Comment (leandroalemao) [inline app/views/api/guidance/view-and-update-participant-data.md:135]: *participant

**Review by leandroalemao — CHANGES_REQUESTED** [review]
All good.. only small comments on typos.
Tested all pages and seems fine apart from I noticed me clicking a lot on browser's "back" button as I haven't found a link to return to the previous page I was.. maybe we can improve the navigation on the guidance pages in the future.

File: app/views/api/guidance/view-and-update-participant-data.md (hunk 21 above)
Comment (cwrw) [reply app/views/api/guidance/view-and-update-participant-data.md]: thanks, all addressed in https://github.com/DFE-Digital/npq-registration/pull/1388/commits/714bd0867b375417e6b3724912d4e33e8e8f00d2

**Review by leandroalemao — APPROVED** [review]
Tks @cwrw

**Review by mooktakim — APPROVED** [review]

**Pull request merged**